subprojects/docs/src/docs/userguide/running-builds/configuration_cache.adoc

```diff
@@ -1048,3 +1048,40 @@ Alternatively, for Kotlin implementations, the Kotlin compiler can be configured
 include::sample[dir="snippets/configurationCache/samConversion/groovy",files="build.gradle[tags=configure]"]
 include::sample[dir="snippets/configurationCache/samConversion/kotlin",files="build.gradle.kts[tags=configure]"]
 ====
+
+[[config_cache:not_yet_implemented:accessing_top_level_at_execution]]
+=== Accessing top-level methods and variables of a build script at execution time
+
+Currently, Gradle cannot run a task restored from the configuration cache if a task action calls a method defined directly in the build script.
+
+In case of the Groovy DSL, the action fails when running from the cache with an error message about method not being found.
+In the following snippet, the task `listFiles` uses such method:
```

Comment:
```suggestion
The following snippet uses a top-level method in the `listFiles` task:
```
Sentence isn't quite grammatically correct -- suggesting rephrase that starts with "The following snippet" to set reader expectations, and that explicitly mentions "top-level" so readers who skip down to an example don't have to backtrack through the whole last paragraph to figure out what's going on.